Cargo.toml

```diff
@@ -10,8 +10,18 @@ name = "crypto"
 
 [features]
 default = []
-
 xchacha20poly1305 = [ "chacha20poly1305" ]
+ed25519 = ["bee-common-derive", "ed25519-dalek", "rand", "signature", "slip10", "zeroize"]
+random = ["rand"]
 
 [dependencies]
+# xchacha20poly1305
 chacha20poly1305 = { version = "0.6.0", optional = true }
+# ed25519
+bee-common-derive = { git = "https://github.com/iotaledger/bee.git", branch = "dev", optional = true }
+ed25519-dalek = { version = "1.0.0-pre.4", features = ["serde"], optional = true }
```

Comment:
@rootmos 
> from https://cr.yp.to/ecdh.html#validate, how does that statement from the creator and the validation issues correlate/coexist?

I hope I understood your question correctly. I'll provide some input and perhaps @Wollac can correct me if I'm wrong.

Ed25519 and Curve25519 are both derived from the same curve, but serve slightly different purposes.
- Ed25519 keypairs are used in sign/verify operation. Each Ed25519 keypair is derived from a seed, and they uniquely identify actors as:
  * `secret key` is used to sign messages.
  * `public key` is used to verify messages.
- Curve25519 keypairs are used in DH key exchanges. Curve25519 keypairs can be deterministically derived from the same seed, but a session `nonce` is derived from them and used to encrypt the channel via some stream-cypher algo.
  * `secret key` in combination with `nonce` is used to encrypt messages.
  * `public key` in combination with `nonce` is used to decrypt messages.

So the key takeaway would be that from the same seed, you can generate both Ed25519 and Curve25519 keypairs. But they shouldn't be confused with eachother, because they serve different purposes (also worth noting that Curve25519 and X25519 are commonly used interchangeably).

Diffie-Hellman key exchange (thus Curve/X25519) doesn't really matter for our case of Distributed Ledger Consensus.

What really matters is that we make sure that message signatures are verified always in a deterministic fashion. And the only way to enforce that is by making `crypto.rs` test vectors (a.k.a. I/O test cases) behaving exactly as proposed on ZIP215.

Here's some useful search results for `ZIP215` test vectors:
- https://gitlab.torproject.org/tpo/core/tor/-/issues/40078 which leads to https://github.com/p-steuer/ossl-eddsa-tests
- https://gist.github.com/hdevalence/93ed42d17ecab8e42138b213812c8cc7